src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockBuilder.cs

```diff
@@ -75,6 +83,12 @@ public TagHelperBlockBuilder(TagHelperBlock original)
         }
 
         /// <summary>
+        /// Text used as a required prefix when matching HTML start and end tags in the Razor source to available 
+        /// tag helpers.
+        /// </summary>
+        public string Prefix { get; set; }
```

Comment:
this class should have a single purpose -- gathering the information necessary to create a `TagHelperBlock`.  this property exists because it's also used on a stack in `TagHelperParseTreeRewriter`.  if you must use it that way, save the `FullTagName` you actually need in `TagHelperParseTreeRewriter.TryRewriteTagHelper()` to match an end tag to its start tag.

btw the current code saves the descriptor's prefix and so matches `<SomeTag>` with `</SomeTag>` if the configured prefix is "some" only because you're doing a case-insensitive match.  will be much more efficient if you save the exact text from the Razor source because the user likely used the same case.  (side note: I'm also not sure if HTML case-insensitivity extends to matching start and end tags with different cases.)